src/registrar/models/domain.py

```diff
@@ -1195,7 +1195,7 @@ def _get_or_create_domain(self):
                     logger.error(e)
                     logger.error(e.code)
                     raise e
-                if e.code == ErrorCode.OBJECT_DOES_NOT_EXIST and self.state != Domain.State.DELETED:
+                if e.code == ErrorCode.OBJECT_DOES_NOT_EXIST and self.state == Domain.State.UNKNOWN:
```

Comment:
**Question:** Was this logic change just so the error was more distinct, or is there a logic/context reasoning behind this change? Sorry if dumb q!

Reply:
This was a bug in the function! It calls `self.dns_needed_from_unknown()` which only accepts states of "unknown". Basically, we'd get this error for domains in the state "ready" trying to run that command

Reply:
OH YES the `django_fsm.TransitionNotAllowed: Can't switch from state 'ready' using method 'dns_needed_from_unknown'` error we were seeing! Thanks for clarifying